components/script/script_module.rs

```diff
@@ -1118,6 +1119,35 @@ unsafe extern "C" fn HostResolveImportedModule(
     unreachable!()
 }
 
+#[allow(unsafe_code, non_snake_case)]
+/// https://tc39.es/ecma262/#sec-hostgetimportmetaproperties
+/// https://html.spec.whatwg.org/multipage/#hostgetimportmetaproperties
+unsafe extern "C" fn HostPopulateImportMeta(
+    cx: *mut JSContext,
+    reference_private: RawHandleValue,
+    meta_object: RawHandle<*mut JSObject>,
+) -> bool {
+    let in_realm_proof = AlreadyInRealm::assert_for_cx(SafeJSContext::from_ptr(cx));
+    let global_scope = GlobalScope::from_context(cx, InRealm::Already(&in_realm_proof));
+
+    // Step 2.
+    let base_url = match (reference_private.to_private() as *const ModuleScript).as_ref() {
+        Some(module_data) => module_data.base_url.clone(),
+        None => global_scope.api_base_url(),
+    };
+
+    rooted!(in(cx) let url_string = JS_NewStringCopyZ(cx, base_url.as_str().as_ptr() as *const i8));
```

Comment:
JS_NewStringCopyZ requires the string to be null-terminated, so we should either use CString here or JS_NewStringCopyN so we can pass the length.